app/helpers/utils.py

```diff
@@ -36,3 +44,47 @@ def get_logging_cfg():
 def init_logging():
     config = get_logging_cfg()
     logging.config.dictConfig(config)
+
+
+def prevent_erroneous_kml(kml_string):
+    kml_string = re.sub(r'on\w*=(".+?"|\'.+?\')', '', kml_string, flags=re.IGNORECASE)
+    kml_string = re.sub(
+        r'(<|&lt;)\s*\bscript\b.*?(>|&gt;).*?(<|&lt;)/\s*\bscript\b\s*(>|&gt;)',
+        ' ',
+        kml_string,
+        flags=re.IGNORECASE | re.DOTALL
+    )
+    return kml_string
+
+
+def validate_content(content):
+
+    def inner_decorator(func):
+
+        @wraps(func)
+        def wrapped(*args, **kwargs):
+            if request.content_type != content:
+                abort(make_error_msg(415, 'Unsupported Media Type'))
+            return func(*args, **kwargs)
+
+        return wrapped
+
+    return inner_decorator
+
+
+def validate_kml_string(kml_string):
+
+    max_file_size = 1024 * 1024 * 2
+
+    if len(kml_string) > max_file_size:
+        error_msg = 'File size exceed %s bytes' % max_file_size
+        logger.error(error_msg)
+        abort(413, error_msg)
+    prevent_erroneous_kml(kml_string)
+    try:
+        ET.fromstring(kml_string)
+    except ParseError as err:
+        logger.error("Invalid kml file %s", err)
+        abort(415, 'Only valid KML file are accepted')
```

Comment:
Here it should return a 400 Bad Request and not 415 Media Type Unsupported. Message should be 'Invalid KML file'